lib/chibi/json.c

```diff
@@ -80,8 +81,12 @@ sexp parse_json_string (sexp ctx, sexp self, sexp str, const char* s, int* i, co
         from = to+1;
         break;
       case 'u':
-        strncpy(utf_tmp, s+to+1, 5);
-        tmp = sexp_make_string(ctx, sexp_make_fixnum(1), sexp_make_character(strtoll(utf_tmp, NULL, 16)));
+        strncpy(utf_tmp, s+to+1, 4);
+        tmp = sexp_make_string(ctx, sexp_make_fixnum(1), sexp_make_character(strtoll(utf_tmp, &utf_cursor, 16)));
+        if( utf_cursor-utf_tmp != 4 || utf_tmp[1] == 'x' || utf_tmp[1] == 'X' || utf_tmp[0] == '-'){
```

Comment:
I think you get unexpected results if the first character is 0 (octal).
Also, strtol will skip over leading whitespace.

nit: the style convention is `if (utf_cursor...`.

Reply:
I didn't think about leading space. I'll arrange something right now.

The octal... hmm not sure. I think if the base is set to 16 it should work fine.